magenta/models/music_vae/js/demo/index.ts

```diff
@@ -1,57 +1,63 @@
 import { MusicVAE, intsToBits, bitsToInts, intsToOneHot } from '../index';
 import * as dl from 'deeplearn'
 
-async function initializeDrums(){
-
-  const mvae:MusicVAE = await new MusicVAE('https://storage.googleapis.com/download.magenta.tensorflow.org/models/music_vae/dljs/drums_small_hikl').initialize();
+async function runDrums(){
+  const mvae:MusicVAE = await new MusicVAE(
+      'https://storage.googleapis.com/download.magenta.tensorflow.org/models/music_vae/dljs/drums_small_hikl').initialize();
 
   const drums = [
     [1, 0, 4, 0, 1, 0, 4, 0, 1, 0, 4, 0, 1, 0, 4, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 4, 0],
     [15, 0, 4, 4, 68, 0, 23, 0, 13, 0, 100, 0, 39, 64, 52, 0, 143, 0, 4, 4, 68, 0, 23, 0, 13, 0, 100, 0, 7, 0, 36, 0],
     [6, 0, 4, 0, 4, 0, 23, 0, 5, 0, 4, 0, 22, 0, 13, 0, 5, 0, 4, 0, 22, 0, 5, 0, 5, 0, 22, 0, 20, 0, 21, 0],
     [0, 0, 0, 0, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]];
 
-  const drumsInput: [number[][], number[][], number[][], number[][]] =
+  const drumsInput: number[][][] =
     [intsToBits(drums[0], 10), intsToBits(drums[1], 10), intsToBits(drums[2], 10), intsToBits(drums[3], 10)];
 
   document.getElementById('drums-inputs').innerHTML = drums.map(d => d.toString()).join('<br>');
 
   let start = Date.now();
 
-  let interp = await mvae.interpolate(drumsInput, 3);
-  document.getElementById('drums-interp-time').innerHTML = ((Date.now() - start) / 1000.).toString() + 's';
-  let interpResults: number[][] = [];
-  for (let i = 0; i < interp.shape[0]; i++) {
-    let bits: Uint8Array[] = [];
-    for (let j = 0; j < interp.shape[1]; j++) {
-      const r: dl.Array3D  = dl.slice3d(interp, [i, j, 0], [1, 1, interp.shape[2]]);
-      bits.push(r.toInt().dataSync() as Uint8Array);
+  dl.tidy(() => {
+    let interp = mvae.interpolate(dl.tensor3d(drumsInput), 3);
+    document.getElementById('drums-interp-time').innerHTML = ((Date.now() - start) / 1000.).toString() + 's';
```

Comment:
if you want more accurate timing, use performance.now()

Reply:
Done.